go/private/rules/binary.bzl

```diff
@@ -118,23 +114,34 @@ def _go_tool_binary_impl(ctx):
     name = ctx.label.name
     if sdk.goos == "windows":
         name += ".exe"
-    out = ctx.actions.declare_file(name)
 
-    command_tpl = ("{go} tool compile -o {out}.a -I {goroot} -trimpath=$PWD $@ && " +
-                   "{go} tool link -o {out} -L {goroot} {out}.a && " +
-                   "rm {out}.a")
-    command = command_tpl.format(
-        go = shell.quote(sdk.go.path),
-        goroot = shell.quote(sdk.root_file.dirname),
-        out = shell.quote(out.path),
+    cout = ctx.actions.declare_file(name + ".a")
+    cargs = ctx.actions.args()
+    cargs.add_all(["tool", "compile"])
+    cargs.add("-o", cout)
+    cargs.add("-I", sdk.root_file.dirname)
```

Comment:
It doesn't actually seem like the `-I` flag is necessary, so let's take it out.

Reply:
Are we sure about that? It means we're doing two things in one CL; A pure refactor and some cleanup.

Reply:
It's pretty low risk, but I'm fine with splitting it into two CLs.

The `-I` and `-L` flags are actually wrong: they should point to `$GOROOT/src` and `$GOROOT/pkg/${GOOS}_${GOARCH}`. But it doesn't seem like the compiler or linker need to be told where to find packages in `GOROOT`, so it never caused problems.